ffi/diplomat/build.rs

```diff
@@ -0,0 +1,14 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+fn main() {
+    if std::env::var("ICU4X_FFI_BAKED_ROOT").is_err() {
+        // Empty data generated with
+        // cargo run -p icu_datagen --features bin,icu_segmenter -- --format mod --use-separate-crates --keys none --out empty_bake
+        println!(
+            "cargo:rustc-env=ICU4X_FFI_BAKED_ROOT={}/empty_bake",
```

Comment:
issue: we shouldn't break the ability to build this crate without build scripts working; there are a bunch of build environments that are bad at running build scripts and ideally this should be something completely unnecessary to run 99% of the time.

I guess we can just ask people using strange build systems to just set this env var, but idk.


I was under the impression baked data was going to be behind a feature?

Reply:
Yeah this is why the GN CI is failing. I was hoping we wouldn't need to feature-gate.

Reply:
I don't immediately see why we need a build.rs to do this computation? Can we use [std::option_env](https://doc.rust-lang.org/std/macro.option_env.html) instead?

Reply:
`include!` requires a string literal argument, which `core::env!("FOO")` is, and `core::option_env!("FOO").unwrap_or_else("bar")` isn't.

Reply:
I don't want to add another dependency; can we just have a feature and doc the feature as "only works with env var set"?

Reply:
It'd be nice if empty-by-default worked, because then that can be used to do data slicing without changing client code.

Reply:
Also, feature documentation is not very discoverable.

Reply:
Proc macro is better than build.rs because GN supports proc macros but not build.rs

Reply:
Clients would just have to pass in `env=empty.rs` for that to work. I don't consider that a huge change given that data slicing is anyway a weird cyclic part of the build with baked data.

Reply:
I would like it if we got automatic DCE-based data slicing in Blaze

Reply:
Ah, I see. We could, though that's going to become a problem again if combinators are used.